src/radical/pilot/agent/launch_method/base.py

```diff
@@ -236,7 +235,7 @@ def get_partitions(self):
     def _create_arg_string(self, args):
 
         if args:
-            return ' '.join([shlex.quote(arg) for arg in args])
+            return ' '.join([ru.sh_quote(arg) for arg in args])
```

Comment:
I saw this new method from src/radical/utils/shell.py #https://github.com/radical-cybertools/radical.utils/commit/238b0666656e6ac9d88023e0ef4cea628153dcd0 but I am not sure how this will work on a fresh new installation, we haven't done any new release of radical utils after that.
So this method may not be found, if we merge this. 

I don't know how the versioning works exactly, but I can test, if it makes any sense.

Reply:
You would install RU from the source tree:
```sh
$ # prepare a virtual environment 
$ virtualenv -p python3 ve3
$ . ve3/bin/activate

$ # install RP
$ git clone git@github.com:radical-cybertools/radical.pilot.git
$ cd radical.pilot
$ git checkout fix/issue_2511
$ pip install .

$ # install a suitable version of RU
$ cd ..
$ git clone git@github.com:radical-cybertools/radical.utils.git
$ cd radical.utils
$ git checkout feature/sh_quote
$ pip uninstall -y radical.utils
$ pip install .
```

The resulting stack should have compatible versions of RU and RP.

Reply:
btw `pip` allows to install directly from GitHub and pick a certain branch/commit
```
pip install git+https://github.com/radical-cybertools/radical.utils.git@devel
```

Reply:
I think I didn't explain the question well. 
 I know how to test this patch by installing RP and RU in common env with feature/fix branches. 
 
 The problem I am trying to highlight is we might be using a older RU dependency  i.e. we might need to update setup.py of RP
I checked out fix/issue_2511 of RP and it is using radical.utils : 1.11.1. 

```
radical-stack 

  python               : /home/workstation/.local/share/virtualenvs/test5env/bin/python3
  pythonpath           : 
  version              : 3.6.15
  virtualenv           : /home/workstation/.local/share/virtualenvs/test5env

  radical.gtod         : 1.6.7
  radical.pilot        : 1.11.2-v1.11.2-59-ga49383992@fix-issue_2511
  radical.saga         : 1.11.1
  radical.utils        : 1.11.1
```

The question is shouldn't this be a new release of RU like RU 1.11.2 and we update the dependency requirement in RP's setup.py 
Currenly it is radical.utils>=1.8.4

Reply:
We update dependency version(s) when upcoming new feature depends on newly introduced feature in dependencies, and that is coming in new releases. But these dependencies are for releases only (master branches).

In your case you required to install RU (and RS if it is needed) manually to test your changes, thus
```
pip install git+https://github.com/radical-cybertools/radical.utils.git@devel
pip install git+https://github.com/radical-cybertools/radical.pilot.git@fix/issue_2511
```